Tests/Functional/Utilities.swift

```diff
@@ -118,21 +118,7 @@ func executeSwiftBuild(_ chdir: String, configuration: Configuration = .Debug, p
     // create special conditions in swift-build for swiftpm tests.
     env["IS_SWIFTPM_TEST"] = "1"
 #if Xcode
-    switch getenv("SWIFT_EXEC") {
-    case "swiftc"?, nil:
-        //FIXME Xcode should set this during tests
-        // rdar://problem/24134324
-        let swiftc: String
-        if let base = getenv("XCODE_DEFAULT_TOOLCHAIN_OVERRIDE")?.chuzzle() {
-            swiftc = Path.join(base, "usr/bin/swiftc")
-        } else {
-            swiftc = try popen(["xcrun", "--find", "swiftc"]).chuzzle() ?? "BADPATH"
-        }
-        precondition(swiftc != "/usr/bin/swiftc")
-        env["SWIFT_EXEC"] = swiftc
-    default:
-        fatalError("HURRAY! This is fixed")
-    }
+    env["SWIFT_EXEC"] = figure_out_SWIFT_EXEC()
```

Comment:
I pushed a different fix for this last night, we will now respect TOOLCHAINS in the environment... if you set TOOLCHAINS for the tests (either by launching Xcode with it, or by changing the environment of the scheme) then that works.

Let's chat today to figure out if there is more we need to do here...